SwiftCheck/Test.swift

```diff
@@ -6,104 +6,204 @@
+/// uses `Testable` early and often in functions and operators to enable a high level of nesting
+/// of framework primitives and an even higher level of genericity in the interface.  By default
+/// SwiftCheck provides `Testable` instances for `Bool`, `Property`, `Prop`, and several other
+/// internal framework types.  Practically, this means any assertions you could make in `XCTest`
+/// will work immediately with the framework.
+
+// MARK: - Quantifiers
+
+/// Below is the method all SwiftCheck properties are based on, `forAll`.  `forAll` acts as a
+/// "Quantifier", i.e. a contract that serves as a guarantee that a property holds when the given
+/// testing block returns `true` or truthy values, and fails when the testing block returns `false`
+/// or falsy values.  The testing block is usually used with Swift's abbreviated block syntax and
+/// requires type annotations for all value positions being requested.  For example,
+///
+///     + This is "Property Notation".  It allows you to give your properties a name and instructs SwiftCheck to test it.
+///     |                                                                + This backwards arrow binds a property name and a property to each other.
+///     |                                                                |
+///     v                                                                v
+///     property("The reverse of the reverse of an array is that array") <- forAll { (xs : Array<Int>) in
+///	        return
+///	            (xs.reverse().reverse() == xs) <?> "Reverse on the left"
+/// 	        ^&&^
+///	 	        (xs == xs.reverse().reverse()) <?> "Reverse on the right"
+///     }
+///
+/// Why require types?  For one, Swift cannot infer the types of local variables because SwiftCheck
+/// uses highly polymorphic testing primitives.  But, more importantly, types are required because 
+/// SwiftCheck uses them to select the appropriate `Gen`erators and shrinkers for each data type
+/// automagically by default.  Those `Gen`erators and shrinkers are then used to create 100 random
+/// test cases that are evaluated lazily to produce a final result.
+
+// MARK: - Going Further
+
+/// As mentioned before, SwiftCheck types do not exist in a bubble.  They are highly compositional
+/// and flexible enough to express unique combinations and permutations of test types.  Below is a
+/// purely illustrative example utilizing a significant portion of SwiftCheck's testing functions:
+///
+///     /// This method comes out of SwiftCheck's test suite.
+///     
+///    `SetOf` is called a "Modifier Type".  To learn more about them see `Modifiers.swift`---+
+///                                                                                           |
+///                                                                                           v
+///     property("Shrunken sets of integers don't always contain [] or [0]") <- forAll { (s : SetOf<Int>) in
```

Comment:
> Added a few more comments up there... but it still leaves a lot of questions – such as "what is shrinkOne and removes and take and drop and when would I use these? 

You wouldn't.  They're all `private` :smile: 

> Also I'd really like to see how to use withFail and whenComplete but I can't find any examples.

Yes, that is certainly lacking  I'll focus there next.